akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -117,6 +120,51 @@ object GraphStages {
     }
   }
 
+  private class TerminationWatcher[T] extends GraphStageWithMaterializedValue[FlowShape[T, T], Future[Unit]] {
+    val name = "TerminationWatcher"
+    val in = Inlet[T](name + ".in")
+    val out = Outlet[T](name + ".out")
+    override def initialAttributes = Attributes.name(name)
+    override val shape = FlowShape(in, out)
+
+    override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, Future[Unit]) = {
+      trait SharingTerminationFuture {
```

Comment:
without this there is warning
[warn] /Users/alexandergolubev/Projects/akka/akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala:159: reflective access of structural type member value finishFuture should be enabled
[warn] by making the implicit value scala.language.reflectiveCalls visible.
[warn] This can be achieved by adding the import clause 'import scala.language.reflectiveCalls'
[warn] or by setting the compiler option -language:reflectiveCalls.
[warn] See the Scala docs for value scala.language.reflectiveCalls for a discussion
[warn] why the feature should be explicitly enabled.
[warn]       (logic, logic.finishFuture)
[warn]                     ^
[info] /Users/alexandergolubev/Projects/akka/akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala:159: method invocation uses reflection
[info]       (logic, logic.finishFuture)
[info]                     ^
[warn] one warning found